lib/block-supports/index.php

```diff
@@ -37,16 +38,26 @@ function gutenberg_apply_block_supports( $block_content, $block ) {
 
 	$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
 	// If no render_callback, assume styles have been previously handled.
-	if ( ! $block_type || ! $block_type->render_callback ) {
+	// Custom CSS is also a server-side hook even for static blocks.
+	$has_custom_css_support = false;
+	if ( property_exists( $block_type, 'supports' ) ) {
+		$has_custom_css_support = gutenberg_experimental_get( $block_type->supports, array( 'customCSS' ), false );
+	}
+	
+	if ( 
+		! ( $block_type && $block_type->render_callback ) &&
+		! $has_custom_css_support
+	) {
```

Comment:
Regardless of the solution in the previous comment, this piece here is definitely something to address: all the other transformations after the early return are for dynamic blocks, and `gutenberg_apply_custom_css_support` is the elusive exception. Static-friendly hooks and dynamic-only hooks should be invoked in clearly different places.